### ISSUE TITLE
Valid diff throws ClassCastException when applied in jvm Clojure

### ISSUE THREAD

**Issue by kthu** [robinheghan/differ#1]
First of all - thank you for differ. We have been using it for a while to store the state of a ClojureScript app to a Clojure server, and its been good. 

However - sometimes we have been seeing exceptions when trying to apply the diffs on the server side. 

After having a closer look at what is happening, it seems changing a value in a map from `nil` to a vector will crash when applying `patch` in Clojure.

For example:

```clojure
(let [new  {:words nil}
      old  {:words ["Bah"]}
      diff (differ/diff old new)]
    (= new (differ/patch old diff)))
```

Will return `true` when evaluated in ClojureScript, but throw a ClassCastException when evaluated in Clojure.

**Comment by robinheghan** [comment]
Interesting! Thank you for this, will have a look later tonight :)

**Comment by kthu** [comment]
Great! 

Please note that I confused myself by having `new` before `old` in the let binding :)  The problem is going from a vector to `nil` (as demonstrated), not the other way around (as explained). Sorry for this.

**Comment by robinheghan** [comment]
I believe this is now fixed in 0.3.3 :)

**Comment by kthu** [comment]
Tusen takk!
